libs/wingsdk/src/simulator/simulator.ts

```diff
@@ -589,8 +608,18 @@ export class Simulator {
     const ResourceType = require(typeInfo.sourcePath)[typeInfo.className];
     const resourceObject = new ResourceType(resolvedProps, context);
 
-    // go ahead and initialize the resource
-    const attrs = await resourceObject.init();
+    const resourceStateDir = this.getResourceStateDir(resourceConfig.path);
+    let attrs: any;
+    if (existsSync(resourceStateDir)) {
+      attrs = await resourceObject.init(resourceStateDir);
+    } else {
+      // if the resource state directory doesn't exist, initialize the resource without state
```

Comment:
I think the robustness should come from the resource implementation and not here so I would always make the directory and pass it down to the resource, and let the resource decide on the behavior if there is no state yet.

This will force the resource to handle partial state better.

It will also allow resources to continuously persist state if they want but that means they need to know the statedir upon initialization even if it's initially empty.

Reply:
OK